runtime/src/main/java/org/corfudb/runtime/CheckpointWriter.java

```diff
@@ -335,32 +343,31 @@ public int appendObjectState(Stream<? extends Map.Entry<?, ?>> entryStream) {
             /* Need to check the size of the compressed buffer. inputByteBuffer and
                inputBuffer are the same buffer.
              */
+            ByteBuf inputBuffer = Unpooled.buffer();
             smrPutEntry.serialize(inputBuffer);
-            ByteBuffer compressedBuffer =
-                    rt.getParameters()
-                            .getCodecType()
-                            .getInstance().compress(ByteBuffer.wrap(inputBuffer.array(),
-                            inputBuffer.readerIndex(), inputBuffer.writerIndex()));
-
-            numBytesPerCheckpointEntry += compressedBuffer.limit();
+            numBytesPerUncompressedCheckpointEntry += smrPutEntry.getSerializedSize();
+            int compressedBufferSize = getCompressedBufferSize(inputBuffer);
+            numBytesPerCheckpointEntry += compressedBufferSize;
+            inputBuffer.clear();
```

Comment:
One thing I learned from the heap dump is that netty ByteBuf clear() only resets reader and writer indexes and does not release the underlying memory. To release the memory we should call release() instead of clear(). But calling release() will not reuse the buffer anymore and introduces more memory allocation overhead which needs to be evaluated.

Reply:
Hmm I think we can leave it as is and since we have the uncompressedSize check which is quite conservative, we should be ok to have the inputBuffer hanging around without releasing it.

Reply:
Hmm... maybe it's best to leave it as is for now. With the conservative uncompressed size check, we should be good to go even with the inputBuffer hanging around

Reply:
I'm not saying we need to address this right now. This is some extra memory we can put in our pocket for now in case we need it in the future.

Reply:
Ah yes, makes sense